README-FRONT-END-DEV.md

```diff
@@ -0,0 +1,204 @@
+* Can you **login to your CFE using your Zooniverse account?** 
+  * i.e. Have you configured your CFE to connect to the Panoptes auth systems? Check your environment, Panoptes App ID, and the valid Callback URLs in the Panoptes OAuth2 Provider/Doorkeeper (links above).
+  * Reminder: on your first login to your CFE, you should be asked to _authorise the Panoptes App._
+* Are you the owner or collaborator of the Project?
+  * You cannot add Transcriptions to Subjects belonging to a Project where you're not an owner/collaborator.
+
+Be aware that the Messenger service has different URLs for its `staging` and `production` settings.
+* `staging` - `https://messenger-staging.zooniverse.org/`
+* `production` - TBC
+
+## 1. Register a Project
+
+Before you start adding Transcriptions to Subjects, you first need to register the Panoptes Project that the Subjects belong to.
+
+This uses the `POST /projects` functionality of the Messenger service, and should only be done **once.** Attempting to register the same project multiple times will net you a bunch of errors.
+
+Here's an example of registering (on `staging`) Project 1651, aka `darkeshard/transfromers`:
+
+```
+import apiClient from 'panoptes-client/lib/api-client.js';
+
+var url = 'https://messenger-staging.zooniverse.org/projects/' +
+          '?slug=' + encodeURIComponent('darkeshard/transformers');  //OK, honestly, I'm not sure if this ?slug part is necessary, but my personal notes says yes? (@shaun.a.noordin 20170630)
+
+var body = JSON.stringify({  //Reminder: turn your JSON objects into strings
+  'data': {
+    'attributes': {
+      'slug': 'darkeshard/transformers'
+    }
+  }
+});
+
+const opt = {
+  method: 'POST',
+  mode: 'cors',
+  headers: new Headers({
+    'Authorization': apiClient.headers.Authorization,
+    'Content-Type': 'application/json',
+  }),
+  body: body,
+};
+
+console.log('REGISTER PROJECT: start');
+
+fetch(url, opt)
+.then((response) => {
+  if (response.status < 200 || response.status > 202) { return null; }
+  return response.json();
+})
+.then((json) => {
+  if (json && json.data) {
+    console.log('REGISTER PROJECT: completed.');
+  }
+})
+.catch((err) => {
+  ...
+});
+
+```
+
+Expected Results:
+* If successful, you should see the `darkeshard/transformers` Project listed when you access `GET /projects/1651`.
+
+Common Problems:
+* If you're seeing 401 Unauthorized errors, chances are you're either:
+  * Not logged into Panoptes.
+  * Logged into `staging` instead of `production` or vice versa.
+  * Logged into an account that's not the owner/collaborator of the Project.
+* Strangely, we've also experienced 401 errors when the `body` string isn't properly structured. Double check your `body`: it should just be data->attributes->slug.
```

Comment:
I'd prefer if you find unexpected or incorrect-seeming statuses/errors that you open an issue rather than include warnings like this. I think it might be confusing and I'd rather fix the issue if one exists.